File: backend/src/Repositories/PageRepository.cs
using backend.Models;
using backend.Models.DTO_s;
using database;
using Microsoft.EntityFrameworkCore;

namespace backend.Repositories;

public class PageRepository
{
    private AppDbContext _context;

    public PageRepository(AppDbContext context)
    {
        _context = context;
    }

    public async Task CreatePage(Page page)
    {
        await _context.Pages.AddAsync(page);
        await _context.SaveChangesAsync();
    }

    public async Task<List<ListPageByUserDTO>> FindPagesByUserId(Guid userId)
    {
        var pages = await _context.Pages
            .Where(p => p.User.Id == userId)
            .Select(p => new ListPageByUserDTO(p.Id, p.Title, p.Description))
            .ToListAsync();

        return pages;
    }

    public async Task<Page> FindPageById(Guid pageId)
    { 
        var page = await _context.Pages.FindAsync(pageId);
        return page;
    }
    
    public async Task UpdatePage(Page page)
    {
        _context.Pages.Update(page);
        await _context.SaveChangesAsync();
    }

    public async Task DeletePage(Page page)
    {
        _context.Pages.Remove(page);
        await _context.SaveChangesAsync();
    }
}

File: mytree/src/main/java/com/mytree/mytree/jpa/PageRepository.java
package com.mytree.mytree.jpa;

import com.mytree.mytree.model.DTOs.PageDTO;
import com.mytree.mytree.model.Page;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PageRepository extends JpaRepository<Page, Integer> {

    @Query("SELECT new com.mytree.mytree.model.DTOs.PageDTO(p.id, p.title, p.description)" +
            "FROM Page p " +
            "WHERE user.id = :userId")
    List<PageDTO> findPagesByUserId(@Param("userId") Integer userId);
}
